addons/hr_holidays/models/hr_leave_allocation.py

```diff
@@ -168,11 +168,16 @@ def _compute_description(self):
                 title = allocation.sudo().private_name
                 if allocation.env.context.get('is_employee_allocation'):
                     if allocation.holiday_status_id:
-                        allocation_duration = allocation.number_of_days_display if allocation.type_request_unit != 'hour' else allocation.number_of_hours_display
+                        if allocation.type_request_unit == 'hour':
+                            allocation_duration = allocation.number_of_days * (allocation.employee_id.sudo().resource_calendar_id.hours_per_day or HOURS_PER_DAY)
```

Comment:
I think you can fallback with the company's calendar before the fallback Hours_per_day and you should add a test :angel: